### PR TITLE
feat: json field filtering [DHIS2-10756]

### PR DESCRIPTION
**Pull request by gnespolino**
This PR supersedes https://github.com/dhis2/dhis2-core/pull/7746
All code from Squiggly library was imported and refactored under "dhis-support-json-fltering" module.
To enable this feature on an endpoint it's enough to implement a FieldFilterCustomizer, and specify URI(s) patterns to enable and a function to customize filters (See PagingWrapperFieldCustomizer). 
Currently, only new Tracker export endpoints have this feature enabled and this PR does not affect any already existing endpoint.

### CODE REVIEW

File: dhis-2/dhis-support/dhis-support-json-filtering/src/main/java/org/hisp/dhis/commons/jsonfiltering/JsonFiltering.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2004-2004-2020, University of Oslo
```
Comment (enricocolasante) [inline dhis-2/dhis-support/dhis-support-json-filtering/src/main/java/org/hisp/dhis/commons/jsonfiltering/JsonFiltering.java:2]: Wrong dates on copyright

File: dhis-2/dhis-support/dhis-support-json-filtering/src/main/java/org/hisp/dhis/commons/jsonfiltering/bean/BeanInfo.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2004-2004-2021, University of Oslo
```
Comment (enricocolasante) [inline dhis-2/dhis-support/dhis-support-json-filtering/src/main/java/org/hisp/dhis/commons/jsonfiltering/bean/BeanInfo.java:2]: Same

File: dhis-2/dhis-support/dhis-support-json-filtering/src/main/java/org/hisp/dhis/commons/jsonfiltering/bean/BeanInfoIntrospector.java [hunk 3]
```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2004-2004-2020, University of Oslo
```
Comment (enricocolasante) [inline dhis-2/dhis-support/dhis-support-json-filtering/src/main/java/org/hisp/dhis/commons/jsonfiltering/bean/BeanInfoIntrospector.java:2]: Same

**Review by enricocolasante — APPROVED** [review]
Overall is really good.
Some copyright's dates are messed up.

I would suggest to double check if we have any e2e tests covering this part and if not planning to create them

**Review by ameenhere — APPROVED** [review]

**Review by stian-sandvold — APPROVED** [review]

**Pull request merged**